src/QMCHamiltonians/tests/test_ion_derivs.cpp

```diff
@@ -983,6 +983,266 @@ TEST_CASE("Eloc_Derivatives:proto_sd_noj", "[hamiltonian]")
 #endif
 }
 
+TEST_CASE("Eloc_Derivatives:proto_sd_wj", "[hamiltonian]")
+{
+  app_log() << "========================================================================================\n";
+  app_log() << "========================================================================================\n";
+  app_log() << "====================Ion Derivative Test:  Prototype Single Slater+Jastrow===========\n";
+  app_log() << "========================================================================================\n";
+  app_log() << "========================================================================================\n";
+  using RealType  = QMCTraits::RealType;
+  using ValueType = QMCTraits::ValueType;
+
+  Communicate* c = OHMMS::Controller;
+
+  const SimulationCell simulation_cell;
+  auto ions_ptr = std::make_unique<ParticleSet>(simulation_cell);
+  auto elec_ptr = std::make_unique<ParticleSet>(simulation_cell);
+  auto &ions(*ions_ptr), elec(*elec_ptr);
+
+  //Build a CN test molecule.
+  create_CN_particlesets(elec, ions);
+  //////////////////////////////////
+  /////////////////////////////////
+  //Incantation to read and build a TWF from cn.wfnoj//
+  Libxml2Document doc2;
+  bool okay = doc2.parse("cn.wfj.xml");
+  REQUIRE(okay);
+  xmlNodePtr root2 = doc2.getRoot();
+
+  WaveFunctionComponentBuilder::PSetMap particle_set_map;
+  particle_set_map.emplace("e", std::move(elec_ptr));
+  particle_set_map.emplace("ion0", std::move(ions_ptr));
+
+  WaveFunctionFactory wff(elec, particle_set_map, c);
+  HamiltonianFactory::PsiPoolType psi_map;
+  psi_map.emplace("psi0", wff.buildTWF(root2));
+
+  TrialWaveFunction* psi = psi_map["psi0"].get();
+  REQUIRE(psi != nullptr);
+  //end incantation
+
+  TWFFastDerivWrapper twf;
+
+  psi->initializeTWFFastDerivWrapper(elec, twf);
+  SPOSet::ValueVector values;
+  SPOSet::GradVector dpsi;
+  SPOSet::ValueVector d2psi;
+  values.resize(9);
+  dpsi.resize(9);
+  d2psi.resize(9);
+
+  HamiltonianFactory hf("h0", elec, particle_set_map, psi_map, c);
+
+  QMCHamiltonian& ham = create_CN_Hamiltonian(hf);
+
+  //This is already defined in QMCHamiltonian, but keep it here for easy access.
+  enum observ_id
+  {
+    KINETIC = 0,
+    ELECELEC,
+    IONION,
+    LOCALECP,
+    NONLOCALECP
+  };
+
+  using ValueMatrix = SPOSet::ValueMatrix;
+
+  int IONINDEX = 1;
```

Comment:
All capital is for compile-time consts.
`const int ion_index =1;`

Reply:
Actually, I'll eliminate this.  Doesn't appear to be used at all.